cg/services/run_devices/pacbio/housekeeper_service/pacbio_houskeeper_service.py

```diff
@@ -86,13 +94,22 @@ def _create_bundle_info(self, file_path: Path, parsed_metrics: PacBioMetrics) ->
         if self._is_file_type_smrt_cell(file_path):
             bundle_name: str = parsed_metrics.dataset_metrics.cell_id
         else:
-            tags.append(parsed_metrics.dataset_metrics.sample_internal_id)
-            bundle_name: str = parsed_metrics.dataset_metrics.sample_internal_id
+            sample_internal_id: str = self._get_sample_internal_id_from_file(
+                file_path=file_path, parsed_metrics=parsed_metrics
+            )
+            tags.append(sample_internal_id)
+            bundle_name: str = sample_internal_id
         return PacBioFileData(
             bundle_name=bundle_name,
             file_path=file_path,
             tags=tags,
         )
 
+    def _get_sample_internal_id_from_file(
+        self, file_path: Path, parsed_metrics: PacBioMetrics
+    ) -> str:
+        barcode: str = file_path.name.split(".")[-2]
+        return self._get_sample_id_from_barcode(barcode=barcode, metrics=parsed_metrics)
```

Comment:
Just to double check - the samples are loaded onto the machine with the internal id rather than sample name?

Reply:
Yes! That Id will come from LIMS